.travis.yml

```diff
@@ -4,17 +4,77 @@ os:
 
 dist: trusty
 
-language : c
+language: c
 
 compiler:
   - clang
   - gcc
+matrix:
+  include:
+    - compiler: "gcc"
+      os: linux
+      env:
+        - PLATFORM=linux BITS=64 HOST=x86_64
+        # TODO do `check` or `check_hw` here too, not just `check_sw` (as part of `coverage`)
+        - CHECK_RULE=coverage GCOV=1
+        - PKG_RULE=gzip
+        - COVERITY_SCAN_PROJECT_NAME="$TRAVIS_REPO_SLUG"
+        - COVERITY_SCAN_BRANCH_PATTERN="travis-ci"
+        - COVERITY_SCAN_NOTIFICATION_EMAIL="robin.getz@analog.com"
+        - COVERITY_SCAN_BUILD_COMMAND_PREPEND="mkdir build && cd build && cmake .."
+        - COVERITY_SCAN_BUILD_COMMAND="make"
+        # The COVERITY_SCAN_TOKEN
+        - secure: "QQt2l9IVo0xXWLx7Xqgz/4Iezj7YroY3MKfmG+OhtzkMRx4prhx32/07JMNAcYx6mQrccyZkBNzpYuXlfIZjFL3xDxDj5+iKv5XXpJbRFQGhc5xZtAlzOIqHNMcFc0Aj8J2mZwKfSaDnBdf+nMgKcmn46pYbpJOmPe9iEfeLvvg="
+
+addons:
+  artifacts: true
 
 before_install:
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install --build-from-source libusb ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install --build-from-source libusb; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get -qq update ; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install -y doxygen; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get install -y libaio-dev; fi
```

Comment:
I could do that - but found it easier to debug with every line installing one thing. I know it looks kind of fugly- but there are limits in yaml (no multi-line), that cause issues as well. I would prefer to leave it - unless you feel strongly about it.